projects/todo-library/src/lib/todo/services/todo.service.ts

```diff
@@ -1,43 +1,53 @@
 import { Injectable } from '@angular/core';
 import { Todo } from '../types/todo';
-import { BehaviorSubject, map, Observable } from 'rxjs';
+import { BehaviorSubject, forkJoin, map, Observable } from 'rxjs';
+import { TodoApiService } from './todo-api.service';
 
 @Injectable()
 export class TodoService {
-  constructor() {}
+  constructor(private apiService: TodoApiService) {}
 
   todoList$: BehaviorSubject<Todo[]> = new BehaviorSubject<Todo[]>([]);
 
   activeTodoList: Observable<Todo[]> = this.todoList$.pipe(
-    map((todos: Todo[]) => todos.filter((todo: Todo) => !todo.isCompleted))
+    map((todos: Todo[]) => todos.filter((todo: Todo) => !todo.completed))
   );
 
   activeTodosCounter: Observable<number> = this.activeTodoList.pipe(
     map((todos) => todos.length)
   );
 
   completedTodoList: Observable<Todo[]> = this.todoList$.pipe(
-    map((todos: Todo[]) => todos.filter((todo: Todo) => todo.isCompleted))
+    map((todos: Todo[]) => todos.filter((todo: Todo) => todo.completed))
   );
 
+  fetchTodos(): void {
+    this.apiService.getTodos().subscribe((todos) => this.todoList$.next(todos));
+  }
+
   addTodo(title: string): void {
     let currentTodoList = this.todoList$.getValue();
-    let newTodo: Todo = {
+    let newTodo = {
       id: Date.now(),
       title,
-      isCompleted: false,
+      completed: false,
     };
-    currentTodoList.push(newTodo);
-    this.todoList$.next(currentTodoList);
+    this.apiService.postTodo(newTodo).subscribe((data) => {
+      currentTodoList.push(data);
+      this.todoList$.next(currentTodoList);
+    });
   }
 
   deleteTodo(id: number): void {
     let currentTodoList = this.todoList$.getValue();
     let index = currentTodoList.findIndex((todo) => todo.id === id);
+    let todo = currentTodoList[index];
     if (index !== -1) {
-      currentTodoList.splice(index, 1);
+      this.apiService.deleteTodoFromBack(todo.id).subscribe(() => {
+        currentTodoList.splice(index, 1);
+        this.todoList$.next(currentTodoList);
+      });
```

Comment:
в сервисе не должно быть подписок, мы ведь обсуждали это :)
возвращай поток, например

```suggestion
      this.apiService.deleteTodoFromBack(todo.id).pipe(
        tap(() => {
          currentTodoList.splice(index, 1);
          this.todoList$.next(currentTodoList);
        })
      );
```